server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -85,7 +86,12 @@ public WriteableSetting(StreamInput in) throws IOException {
         if (hasFallback) {
             fallback = new WriteableSetting(in);
         }
-        // We are using known types so don't need the parser
+        // Read the parser
+        WriteableSetting parser = null;
```

Comment:
The parser isn't a setting, so this is not the correct type.  At this point all you know about it is that it's a `Writeable` (and also a `Function`).